client/src/components/JurisdictionAdmin/BatchRoundDataEntry/BatchDetails.tsx

```diff
@@ -0,0 +1,472 @@
+  contest: IContest
+  saveBatchResults: (
+    resultTallySheets: IBatchResultTallySheet[]
+  ) => Promise<void>
+  setAreChangesUnsaved: (areChangesUnsaved: boolean) => void
+
+  // Require a key to ensure that the state within this component resets when a different batch is
+  // selected
+  key: string // eslint-disable-line react/no-unused-prop-types
+}
+
+const BatchDetails: React.FC<IBatchDetailsProps> = ({
+  areResultsFinalized,
+  batch,
+  contest,
+  saveBatchResults,
+  setAreChangesUnsaved,
+}) => {
+  const [sheets, setSheets] = useState<IBatchResultTallySheetWithId[]>(
+    (batch.resultTallySheets.length === 0
+      ? [constructEmptySheet(defaultSheetName(1), contest)]
+      : batch.resultTallySheets
+    ).map(sheetWithId)
+  )
+  const tabs = tabsFromSheets(sheets)
+  const [selectedTabId, setSelectedTabId] = useState(tabs[0].id)
+  const [isEditing, setIsEditing] = useState(false)
+
+  const currentSheetIndex = sheets.findIndex(
+    sheet => sheet.id === selectedTabId
+  )
+
+  // ---------- Handlers (start) ----------
+
+  const updateCurrentSheet = async (updatedSheet: IBatchResultTallySheet) => {
+    const updatedSheets = sheets.map((sheet, i) =>
+      i === currentSheetIndex ? { ...updatedSheet, id: sheet.id } : sheet
+    )
+
+    // Update client-side state first for immediate UI feedback
+    setSheets(updatedSheets)
+    await saveBatchResults(updatedSheets.map(sheetWithoutId))
+  }
+
+  const addSheet = async () => {
+    const isSheetNameAlreadyTaken = (sheetName: string) =>
+      sheets.some(sheet => sheet.name === sheetName)
+
+    let incrementToAvoidNamingConflict = 0
+    let newSheetName = defaultSheetName(sheets.length + 1)
+    while (isSheetNameAlreadyTaken(newSheetName)) {
+      incrementToAvoidNamingConflict += 1
+      newSheetName = defaultSheetName(
+        sheets.length + 1 + incrementToAvoidNamingConflict
+      )
+    }
+    const newSheet = sheetWithId(constructEmptySheet(newSheetName, contest))
+    const updatedSheets = [...sheets, newSheet]
+
+    // Update client-side state first for immediate UI feedback
+    setSheets(updatedSheets)
+    setSelectedTabId(newSheet.id)
+    await saveBatchResults(updatedSheets.map(sheetWithoutId))
+  }
+
+  const deleteSheet = async () => {
+    let updatedSheets = sheets.filter((_, i) => i !== currentSheetIndex)
+    if (updatedSheets.length === 1) {
+      // If we're dropping back to 1 sheet, reset the name of that sheet back to the default
+      updatedSheets = [{ ...updatedSheets[0], name: defaultSheetName(1) }]
+    }
+
+    // Update client-side state first for immediate UI feedback
+    setSheets(updatedSheets)
+    setSelectedTabId(
+      // Auto-select the next sheet (or last sheet if none)
+      updatedSheets[Math.min(currentSheetIndex, updatedSheets.length - 1)].id
+    )
+    await saveBatchResults(updatedSheets.map(sheetWithoutId))
+  }
+
+  // ---------- Handlers (end) ----------
+
+  return (
+    <Detail>
+      <H4>{batch.name}</H4>
```

Comment:
Biggest nit of all time: can we baseline align the batch name with the text in the search field?

<img width="422" alt="Screen Shot 2022-10-31 at 11 36 57 AM" src="https://user-images.githubusercontent.com/530106/199084073-604fac90-6b75-4813-b632-594119372cfc.png">

Reply:
https://github.com/votingworks/arlo/pull/1661/commits/c3b9372fc957c60c575a7659161c096934e42804

Reply:
Decided to vertically middle align the batch name and search bar since it was a bit easier to do so precisely:

<img width="400" alt="alignment" src="https://user-images.githubusercontent.com/12616928/199336689-4be42d20-dfe9-45a1-8ad4-722f71d68442.png">

https://github.com/votingworks/arlo/pull/1661/commits/3d03b4a0df4040acdd77bdcd9d1d1d04e5d9ff67

Reply:
That works for me!